modules/mobile/app/controllers/mobile/v0/appointments_controller.rb

```diff
@@ -49,14 +49,18 @@ def cancel
 
       def use_cache?(validated_params)
         # use cache if date range is within +/- 1 year and use_cache is true
-        validated_params[:start_date] >= one_year_ago.iso8601 &&
+        validated_params[:start_date] >= beginning_of_last_year.iso8601 &&
```

Comment:
I think this PR needs to get rid of one_year_ago entirely. All uses of it should be replaced with beginning_of_last_year. As it's written now, it will fetch from the cache if the start date is >= beginning of last year, but it's not actually fetching records from that time, so the cache won't include records starting at the beginning of last year. The other two places it's used are: 1) setting the default start date value when none is provided by the client 2) indicating the topmost range of the start date we allow to be passed into the proxy to ensure that the cache remains valid for the default range.

Reply:
I was thinking to avoid changing default behavior as to not make the default more expensive then is likely needed to be for the user going on the appointment page without date filters. As it is now, as long as the start and end date are set in params, they can use the cache as early as beginning of last year. Thinking about it more, even if we do get more appointments by broadening the default window, it should only be marginally more expensive since we do paginate. Make the code cleaner and more consistent too.

Reply:
You're right that it'll make the call slightly more expensive. And it's possible that for a power appointments user like judy it might even start causing timeouts. In which case we'd have to revisit the caching strategy. But it's also worth remembering that the app pre-caches appointments when a user logs in, so it shouldn't impact the user most of the time. Speaking of which... I just realized that you also need to update the start date in `modules/mobile/app/workers/mobile/v0/pre_cache_appointments_job.rb`.

But the point I'm making above is that the cache will not be accurate if you don't change all instances of the start date.

Reply:
I see your point now. I updated the default start date to match as well as the start dates of the VCR cassettes. Still working on additional specs to explicitly test the default start date